### PR TITLE
Unify dependencies across packages

### LINKED ISSUE #1: Hoist common dependencies to workspace

### LINKED ISSUE #307: Allow all libfuncs in contracts
The easiest solution is to use casm generated by Scarb. In this case we need to update starknet targets in Scarb.tomls accordingly (`casm=true`).

### LINKED ISSUE #327: Update Cairo VM to 0.8

### LINKED ISSUE #331: Unify cairo versions used in packages, bump to `2.1.0-rc1`

### PR DESCRIPTION
**Pull request by cptartur**
Closes #1 
Closes #307 
Closes #327 
Closes #331

### CODE REVIEW

**Comment by cptartur** [comment]
The build passes but any kind of interaction with starknet fails on declare with this error `CairoResourcesNotContainedInFeeCosts`.

It seems that blockfier introduced some kind of fee validation and we aren't passing it because we are using empty HashSet for resource costs. But I haven't done extensive verification of that so the issue might be somewhere else.

File: crates/forge/src/cheatcodes_hint_processor.rs [hunk 1]
```diff
@@ -785,11 +799,13 @@ mod test {
...
         let cases = [
             (
-                "0x5167b09ea07d236371efa6053537eb2e6163bdb531e7161643a7c1fddfc5814",
+                // TODO verify calculation of this
+                "0x00b0e07d0ab5d68a22072cd5f35f39335d0dcbf1a28fb92820bd5d547c497f33",
                 "target/dev/simple_package_ERC20.casm.json",
             ),
             (
-                "0x192485856ebf42c113825d359a948dacaf526d62bf6c2aa231beffed0fddc3f",
+                // TODO verify calculation of this
```
Comment (piotmag769) [inline crates/forge/src/cheatcodes_hint_processor.rs:807]: TODO

File: crates/forge/src/cheatcodes_hint_processor.rs [hunk 2]
```diff
@@ -785,11 +799,13 @@ mod test {
         // https://github.com/software-mansion/starknet.py/blob/cea191679cbdd2726ca7989f3a7662dee6ea43ca/starknet_py/tests/e2e/docs/guide/test_cairo1_contract.py#L29-L36
         let cases = [
             (
-                "0x5167b09ea07d236371efa6053537eb2e6163bdb531e7161643a7c1fddfc5814",
+                // TODO verify calculation of this
```
Comment (piotmag769) [inline crates/forge/src/cheatcodes_hint_processor.rs:802]: TODO

File: crates/forge/src/cheatcodes_hint_processor.rs [hunk 3]
```diff
@@ -490,15 +491,25 @@ fn declare(
...
     )
     .unwrap_or_else(|err| panic!("Unable to build transaction {err:?}"));
 
     let account_tx = AccountTransaction::Declare(tx);
     let block_context = build_block_context();
-    let _tx_result = account_tx
-        .execute(blockifier_state, &block_context)
-        .context("Failed to execute declare transaction")?;
+    let tx_result = account_tx
+        // FIXME not sure we should be using true or false for these
```
Comment (piotmag769) [inline crates/forge/src/cheatcodes_hint_processor.rs:503]: FIXME

File: crates/forge/src/cheatcodes_hint_processor.rs [hunk 4]
```diff
@@ -490,15 +491,25 @@ fn declare(
     );
     let tx = DeclareTransaction::new(
         starknet_api::transaction::DeclareTransaction::V2(declare_tx),
+        // FIXME incorrectly calculating hashes may be the case of the problems
```
Comment (piotmag769) [inline crates/forge/src/cheatcodes_hint_processor.rs:494]: FIXME

File: crates/cheatable-starknet/src/constants.rs [hunk 5]
```diff
@@ -42,13 +48,25 @@ pub const TEST_ERC20_CONTRACT_CLASS_HASH: &str = "0x1010";
...
+    // blockifier::test_utils::create_for_account_testing
+    let vm_resource_fee_cost = Arc::new(HashMap::from([
+        (constants::N_STEPS_RESOURCE.to_string(), 1_f64),
+        (HASH_BUILTIN_NAME.to_string(), 1_f64),
+        (RANGE_CHECK_BUILTIN_NAME.to_string(), 1_f64),
+        (SIGNATURE_BUILTIN_NAME.to_string(), 1_f64),
+        (BITWISE_BUILTIN_NAME.to_string(), 1_f64),
+        (POSEIDON_BUILTIN_NAME.to_string(), 1_f64),
+        (OUTPUT_BUILTIN_NAME.to_string(), 1_f64),
+        (EC_OP_BUILTIN_NAME.to_string(), 1_f64),
```
Comment (piotmag769) [inline crates/cheatable-starknet/src/constants.rs:60]: I think we should add an issue to investigate if the costs are fine

**Review by piotmag769 — COMMENTED** [review]
We can get rid of blockifier_Felt252 now

**Review by piotmag769 — COMMENTED** [review]
Do we need sierra artifacts generated by Scarb now? cc @cptartur

File: crates/forge/src/cheatcodes_hint_processor.rs [hunk 6]
```diff
@@ -490,15 +477,25 @@ fn declare(
...
+        TransactionHash(StarkFelt::from(1_u32)),
```
Comment (cptartur) [inline crates/forge/src/cheatcodes_hint_processor.rs:481]: Maybe lets just use `TransactionHash::default()` for now. It'll be easier to replace all occurrences of it in the future.

File: crates/forge/src/cheatcodes_hint_processor.rs (hunk 3 above)
Comment (cptartur) [reply crates/forge/src/cheatcodes_hint_processor.rs:489]: I guess we should at least agree on what should be used here and remove the fixme before the merge.

File: crates/forge/src/cheatcodes_hint_processor.rs (hunk 6 above)
Comment (cptartur) [reply crates/forge/src/cheatcodes_hint_processor.rs:481]: And add a corresponding TODO here

File: crates/forge/src/cheatcodes_hint_processor.rs (hunk 2 above)
Comment (cptartur) [reply crates/forge/src/cheatcodes_hint_processor.rs:788]: I think we should use the linked python code to verify that and remove the TODOs. You can create an issue and we can verify in separate PR.

File: crates/cheatable-starknet/src/constants.rs (hunk 5 above)
Comment (Radinyn) [reply crates/cheatable-starknet/src/constants.rs:60]: #362

**Review by cptartur — COMMENTED** [review]
I'd request changes but I cannot on my own PR

File: crates/forge/src/cheatcodes_hint_processor.rs [hunk 7]
```diff
@@ -490,15 +477,25 @@ fn declare(
...
+    let tx_result = account_tx
+        // FIXME not sure we should be using true or false for these
+        .execute(blockifier_state, &block_context, true, true);
+
+    match tx_result {
+        Ok(_) => (),
+        Err(e) => {
+            return Err(anyhow!(format!("Failed to execute declare transaction:\n    {e}")).into())
+        }
+    };
```
[the remark this answers is not in the document]
Comment (Radinyn) [reply crates/forge/src/cheatcodes_hint_processor.rs:497]: I think it's the least confusing way

File: crates/forge/src/cheatcodes_hint_processor.rs (hunk 6 above)
Comment (Radinyn) [reply crates/forge/src/cheatcodes_hint_processor.rs:481]: done

File: crates/forge/src/cheatcodes_hint_processor.rs (hunk 3 above)
Comment (Radinyn) [reply crates/forge/src/cheatcodes_hint_processor.rs]: Leaving it as `true`, `true` as it seems to make sense

#371

File: crates/cheatable-starknet/src/constants.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (Radinyn) [reply crates/cheatable-starknet/src/constants.rs:60]: Starkware uses `1` in testing

[the remark this answers is not in the document]
Comment (Radinyn) [reply crates/cheatable-starknet/src/constants.rs:60]: #362

**Review by piotmag769 — APPROVED** [review]
LGTM

File: crates/forge/src/cheatcodes_hint_processor.rs [hunk 8]
```diff
@@ -490,15 +477,23 @@ fn declare(
...
-        .execute(blockifier_state, &block_context)
-        .context("Failed to execute declare transaction")?;
+    let tx_result = account_tx.execute(blockifier_state, &block_context, true, true);
+
+    match tx_result {
+        Ok(_) => (),
+        Err(e) => {
+            return Err(anyhow!(format!("Failed to execute declare transaction:\n    {e}")).into())
+        }
+    };
```
Comment (cptartur) [inline crates/forge/src/cheatcodes_hint_processor.rs:495]:
```suggestion
    match account_tx.execute(blockifier_state, &block_context, true, true) {
        Ok(_) => (),
        Err(e) => {
            return Err(anyhow!(format!("Failed to execute declare transaction:\n    {e}")).into())
        }
    };
```

`tx_result` isn't being used anywhere else, so we can at least combine these.

**Review by Radinyn — APPROVED** [review]

**Pull request merged**